### DISCUSSION TITLE
Update Trust Wallet connector and other changes

### DISCUSSION THREAD

**Discussion by george-gkas** [rainbow-me/rainbowkit#949] [Ideas]
:wave: Hello, `rainbowkit` team. My name is George, and I'm currently a Frontend Engineer on the Web Extension team of [Trust Wallet](https://trustwallet.com). In the last couple of months, we introduced significant updates to our [web extension](https://trustwallet.com/browser-extension) in line with our values to support impactful Web3 technologies; we would like to contribute by submitting some changes to the Trust Wallet connector. We already did most of the work, and now our QA department is handling the testing procedure. Based on [RainbowKit Contribution Guide](https://github.com/rainbow-me/rainbowkit/blob/main/.github/CONTRIBUTING.md), we open this discussion to let you know about the upcoming changes. Below is an extensive list of those introduced changes:

**Trust Wallet Connector**
- Handled cases where the web extension was installed, but the injected provider was not used by introducing the `installed` property.
- Added `browserExtension` field in `downloadUrls`.
- Added `iconAccent` property.
- Added instructions for web extension flow.
- Fix edge cases where the incorrect injected provider was used when users have multiple injected wallets installed.

**Example apps**
- Removed Alchemy Provider from example apps. The current Alchemy API key needed to be updated, rendering all the examples unusable. We tested using only`publicProviders`, and the examples work as expected. This is a better solution because we rely on something other than third-party API keys that might need to be updated. Here is the list of the example apps that this change applies:
  - `examples/with-create-react-app`
  - `examples/with-next-custom-button`
  - `examples/with-next-siwe-next-auth`
  - `examples/with-next`
  - `examples/with-remix`
  - `examples/with-vite`
- Fixed the issue with ESM imports on `examples/with-remix`, preventing the app from compiling by adding the correct `serverDependenciesToBundle` rule.
- Fixed the following issues introduced by CRA v5 in `examples/with-create-react-app`:
  - [Browser polyfill.](https://github.com/facebook/create-react-app/issues/11756)
  - [Issues with source maps errors and warnings](https://github.com/facebook/create-react-app/discussions/11767).

**Core library**
- Added Trust Wallet in the list returned by `getDefaultWallets`. Trust Wallet extension has reached hundreds of thousands of downloads, and we keep expanding. Many of our users request to connect with Trust Wallet by default, and it is now placed as a fourth choice in the connection UI modal.
<img width="740" alt="Screenshot 2023-01-11 at 2 16 56 PM" src="https://user-images.githubusercontent.com/118829911/211804361-e7bf4fbf-bfd8-4e2a-b5aa-f6ed509dd98e.png">
- Added Trust Wallet download option in mobile view.
<img width="389" alt="Screenshot 2023-01-11 at 2 18 52 PM" src="https://user-images.githubusercontent.com/118829911/211804642-e591e217-5bf6-4286-addc-c793080f2535.png">

**Documentation**
- Added `iconAccent` in the documentation site.

We are open to talks and any questions you might have. Please keep in mind that we will continue to support this connector in the long run. 🚀

**Comment by george-gkas** [comment]
PR submitted: https://github.com/rainbow-me/rainbowkit/pull/950

**Reply by DanielSinclair** [reply]
Addressed all of the issues referenced here with a plan to break up #950
